runtime/common/src/main/kotlin/kotlinx/serialization/json/JsonTreeParser.kt

```diff
@@ -1,85 +1,13 @@
 /*
- * Copyright 2018 JetBrains s.r.o.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- * http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * Copyright 2017-2018 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license.
  */
 
 package kotlinx.serialization.json
 
-import kotlinx.serialization.json.internal.*
-
-class JsonTreeParser internal constructor(private val parser: JsonParser) {
+@Deprecated(level = DeprecationLevel.ERROR, message = "Use Json.parseJson instead")
```

Comment:
To avoid a long deprecation cycle.
The compilation will break, replacement is clear, no need to introduce constructor or `read` method.

Can return it back if you insist.